src/blackmetal/src/communication/Client.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <string>
+#include <variant>
+
+/**
+ * @namespace bm
+ * @brief namespace containing the enums for blackmetal constants.
+ */
+namespace bm
+{
+
+/**
+ * @enum Command
+ * @brief Commands defined in the blackmetal robot documentation.
+ */
+enum class Command {
+	/// Empty command
+	EMPTY,
+	/// Emergency stop
+	EMG_SOPT,
+	/// Normal stop
+	NORMAL_STOP,
+	/// Sets Left and right wheel velocity.
+	/// The velocities are required for this command.
+	SET_LR_WHEEL_VELOCITY,
+	/// Not implemented.
+	NONE_4,
+	/// Not implemented.
+	NONE_5,
+	/// Get left and right wheel velocities.
+	GET_LR_WHEEL_VELOCITY,
+	/// Prepare the controller of left and right wheel.
+	PREPARE_WHEEL_CONTROLLER,
+};
+
+/**
+ * @enum Status
+ * @brief return status of the server.
+ * The robot receives the json string and returns a json string with a specific return status.
+ * This status is than mapped on BlackMetal::Status enum type.
+ */
+enum class Status {
```

Comment:
Header by som rozdelil na Client, ktory iba implementuje komunikaciu a druhy header, ktory parsuje

Reply:
Rozdelil som to na dva subory ale nevidim zmysel to davat do ineho suboru kedze tento client bude pracovat len s tymito enumami a nicim inym. Keby ze to malo byt genericke tak by som to spravil inak. Ako to vidis ty?

Reply:
Osobne mi to nevadi, len z mojej skusenosti odporucam vzdy sa drzat "single-responsibility principle", cize jeden objekt by mal robit len jednu vec. Preto by som to rozdeloval na komunikaciu a na parsovanie.